src/tools/rbd_mirror/InstanceReplayer.cc

```diff
@@ -224,7 +224,10 @@ void InstanceReplayer<I>::remove_peer_image(const std::string &global_image_id,
     // it will eventually detect that the peer image is missing and
     // determine if a delete propagation is required.
     auto image_replayer = it->second;
-    image_replayer->restart(new C_TrackedOp(m_async_op_tracker, nullptr));
+    auto ctx = new LambdaContext([this, image_replayer] (int r) {
+        image_replayer->stop();
+      });
+    image_replayer->restart(new C_TrackedOp(m_async_op_tracker, ctx));
```

Comment:
I don't think I like it. To me it looks like a workaround for a case when `image_replayer->is_finished()` does not work? Then I would rather fix this.

And this change seems not to be explained in the commit log?

Also note, there is a problem with your code, that you do not track the image replayer stop (with `m_async_op_tracker`), so the instance replayer may be shut down while the stop is still in progress.

Reply:
This is not meant to make sure the ImageReplayer "finish" properly but rather to cleanup the image status in some case. I don't recall the exact situation when this happens, I will run the qa script and provide you with more details and will add it in the commit message.

Reply:
Ah in fact I think I allready fixed it with other fixes :) I am dropping this commit then.